### PR TITLE
Should now include all resources from oppgave-repo

### PR DESCRIPTION
**Pull request by NorwegianKiwi**
Should solve issue #387. See also PR #357 and #367, as well as PR #352 and #386.

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ const readmeContext = require.context('onlyFrontmatter!lessonSrc/', true,
   /^\.\/[^\/]*\/[^\/]*\/README(_[a-z]{2})?\.md$/);
 const lessons = getLessons(lessonContext, readmeContext, courseContext);
 
+// Add all resources, i.e. all files in lessonSrc that does not end in '.md':
+require.context('lessonSrc/', true, /\.(?!md$)[^.]+$/);
```
Comment (NorwegianKiwi) [inline src/store.js]: I think perhaps this line is not necessary. It forces webpack to include ALL files that don't end in `.md`, but I think markdown-it-modify-token (setting data-src) combined with the new `attrs`-parameter for html-loader should include/require all resources that are linked to from the lessons. So by deleting this line (require.context) we avoid copying resources that are not linked to (i.e. not used).

Needs to be tested first though.

**Comment by NorwegianKiwi** [comment]
I have an idea to an alternative PR which does more or less what you suggest, but using the [copy-webpack-plugin](https://github.com/kevlened/copy-webpack-plugin). The reason for not using your suggestion with fs, is that (as far as I understand) it wouldn't work with react-hot-dev (where the files are not copied on disk). Anyway, copy-webpack-plugin would probably be even simpler than your suggestion above. I am thinking as long as it is copied to the "correct" place in the dist-folder, it should work.

Again, this would mean that we would copy EVERY non-'.md' file instead of just those that are linked to, i.e. we will have unused files on the site (if there are unused files in the oppgave-repo), but if that is ok, we might be able to simplify.

When I have made the PR, I will make a reference to it here.

**Comment by arve0** [comment]
> The reason for not using your suggestion with fs, is that (as far as I understand) it wouldn't work with react-hot-dev (where the files are not copied on disk).

One could add a watcher, like this one: https://github.com/zont/copy-and-watch

**Comment by NorwegianKiwi** [comment]
As I said, I don't think that would work for react-hot-dev, which doesn't actually copy the files on the disk system, but into memory. Of course, it doesn't HAVE to work for react-hot-dev, but it would make it more impractical for the developers.

**Comment by NorwegianKiwi** [comment]
See https://github.com/kodeklubben/codeclub-viewer/pull/390 for an alternative PR.

**Comment by Skagevang** [comment]
Went for #390

**Pull request closed without merging**